### ISSUE TITLE
--loader-config not working on Node 21

### ISSUE THREAD

**Issue by mmmulani** [sebamarynissen/node-esm-loader#5]
I'm using Node 21.2.0 locally, invoking node-esm-loader like `node --import node-esm-loader/register index.js --loader-config=./loaderConfig.js` but from adding logging to `lookup.js`, notably `process.env.LOADER_CONFIG` is undefined. It seems that env is not passed to the loader process?

I was able to work-around this by doing
```
LOADER_CONFIG=./loaderConfig.js node --import node-esm-loader/register index.js
```

It seems the env is copied before running the import part of the module, poked around in the Node source code but couldn't find where this happened.

**Comment by sebamarynissen** [comment]
Ah yes, when updating the module to work with Node's [module.register](https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options) approach, I noticed that the arguments the process was started with were not passed to the separate thread that the register uses, so I added this workaround with `LOADER_CONFIG`, but probably should've documented it.

I'll have a look if I can get it to work with passing the arguments via the `initialize()` hook.

**Comment by sebamarynissen** [comment]
If you have a look at `register.js`, you see
```js
// On Node 20 the arguments are no longer passed to the loader thread, but 
// environment variables are, so if a file was specified, we transfer it using 
// the LOADER_CONFIG environment variable.
const parsed = yargs(hideBin(process.argv)).help(false).version(false);
const config = parsed.argv['loader-config'];
if (config) {
	process.env.LOADER_CONFIG = config;
}
```
so that's the workaround I used to pass the loader-config. This was working around the problem on my local Windows machine, perhaps that it doesn't on Mac though.

**Comment by sebamarynissen** [comment]
I've done some research and found that actually `--loader-config` does not work anymore either if you use `--experimental-loader node-esm-loader` because the loader also runs in a separate thread that case (it did not do that on earlier Node versions), so `process.argv` was not available there either.

I have found a solution with the `initialize()` hook, but this will only work when using Node's new `register()`. I will probably release a new version of the module as a breaking change where you'll only be able to use it with `register()` as it seems that this is what the loaders api has settled on. It has been backported to Node 18 as well, so it will still work on both 18 and 20 LTS.

**Comment by sebamarynissen** [comment]
Should be fixed now with the release of `0.3.0`.

**Comment by mmmulani** [comment]
awesome! thanks for the fast fix and release
